bundles/org.eclipse.osgi/felix/src/org/apache/felix/resolver/ResolverImpl.java

```diff
@@ -45,7 +77,7 @@ public class ResolverImpl implements Resolver
 
     private final Executor m_executor;
 
-    enum PermutationType {
+    public enum PermutationType {
```

Comment:
It is odd to have the Logger takes enums of this inner type on the ResolverImpl.  Should the enum be moved to Logger?

Reply:
I would rather extract it into an own class then.

Reply:
Done.